Applications/Spire/Source/Ui/TagBox.cpp

```diff
@@ -0,0 +1,513 @@
+#include "Spire/Ui/TagBox.hpp"
+#include <QHBoxLayout>
+#include <QKeyEvent>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/ArrayListModel.hpp"
+#include "Spire/Ui/Box.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/ListItem.hpp"
+#include "Spire/Ui/ListView.hpp"
+#include "Spire/Ui/Tag.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto INPUT_BOX_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(vertical_padding(scale_height(3)));
+    return style;
+  }
+
+  auto LIST_ITEM_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(BackgroundColor(QColor(Qt::transparent))).
+      set(border_size(0)).
+      set(padding(0));
+    return style;
+  }
+
+  auto LIST_VIEW_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(ListItemGap(scale_width(4))).
+      set(ListOverflowGap(scale_width(3))).
+      set(Overflow::WRAP).
+      set(Qt::Horizontal);
+    auto& list_item_rule = style.get(Any() >> is_a<ListItem>());
+    auto list_item_style = LIST_ITEM_STYLE();
+    for(auto& rule : list_item_style.get_rules()) {
+      for(auto& property : rule.get_block()) {
```

Comment:
What is the goal of these nested for loops? Why are you applying a style to every rule instead of applying it to the ```Any()``` selector?